### PR TITLE
feat: integration of vue simulator to main repository

### LINKED ISSUE #3768: Integration of cv-frontend-vue repository with the circuitverse (main) repository : part 1
### tasks to be done:

- [x] Add the cv-frontend-vue repository into main repository with the help of git submodules

- [x] setup the routes for the new simulator

- [x] update esbuild to configure the build of the vue simulator

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by Arnabdaz**
Fixes #3768 

#### Describe the changes you have made in this PR -

- [x] add cv-frontend-vue as git submodule
- [x] add build config for vue simualtor in esbuild.config.js
- [x] add /simualtorvue route for initial integration
- [x] update dockerfile.production for vue simulator
- [x] update .gitpod.yml for vue simulator
- [x] update workflows 
     - docker-publish.yml
     - rubyonrails.yml
- [x] add static controller for vue simualtor
- [x] spec for static controller 

### Screenshots of the changes (If any) -

N/A

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: Dockerfile.production [hunk 1]
```diff
@@ -37,9 +38,24 @@ RUN rm -rf node_modules spec
 
 RUN bundle exec bootsnap precompile --gemfile app/ lib/
```
Comment (tanmoysrt) [inline Dockerfile.production:40]: Add a suitable comment for this build stage

File: spec/controllers/static_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+RSpec.describe StaticController, type: :request do
+  describe "#simulatorvue" do
+    it "renders the simulatorvue template" do
+      get "/simulatorvue"
+
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include(File.read(Rails.public_path.join("simulatorvue", "index.html")))
+    end
+  end
+end
```
Comment (tanmoysrt) [inline spec/controllers/static_controller_spec.rb:16]: One solution is working fine, in all other cases undercover is raising the issue

In the `rails_helper` file, we need to put this line
```ruby
  config.include Devise::Test::ControllerHelpers, type: :controller
```

And the updated specs will be
```ruby
# frozen_string_literal: true

# spec/controllers/static_controller_spec.rb

require "rails_helper"

RSpec.describe StaticController, type: :controller do
  describe '#simulatorvue' do
    it 'renders the simulatorvue template' do
      get :simulatorvue, params: {path: "/"}

      expect(response).to have_http_status(:success)
      expect(response.body).to include(File.read(Rails.public_path.join("simulatorvue", "index.html")))
    end
  end
end
```

**Review by tanmoysrt — CHANGES_REQUESTED** [review]
The Docker Setup is working as expecte in both development and production environment.

File: spec/controllers/static_controller_spec.rb (hunk 2 above)
Comment (Arnabdaz) [reply spec/controllers/static_controller_spec.rb]: Thanks for your help review dog issue is fixed now.

File: Dockerfile.production (hunk 1 above)
Comment (Arnabdaz) [reply Dockerfile.production:41]: I have added a comment can you please review ?

**Review by tanmoysrt — APPROVED** [review]
everything looks fine. Codeclimate duplicate code issue needs to be fixed !

cc @vedant-jain03 @Arnabdaz

File: .github/workflows/rubyonrails.yml [hunk 3]
```diff
@@ -26,13 +26,14 @@ jobs:
...
           fetch-depth: 100
+          submodules: recursive
       - name: Install Ruby and gems
         uses: ruby/setup-ruby@v1 # v1.115.3
         with:
           bundler-cache: true
       - uses: actions/setup-node@v3
         with:
-          node-version: '14'
+          node-version: '16'
```
Comment (vedant-jain03) [inline .github/workflows/rubyonrails.yml:36]: @Arnabdaz have you tested the compatibility with this node version?

Comment (Arnabdaz) [reply .github/workflows/rubyonrails.yml:36]: Yes

**Pull request merged**